### PR TITLE
Bundle images

### PR DESCRIPTION
**Pull request by jepler**
Typical image (for the "turtle_swirl" example, filename **turtle_turtle_swirl py**)

turtle_turtle_swirl py

Re-uses all the dependency chasing code of the code for Learn but generalizes it a bit so it can be used for library examples.

A subsequent pull-request over in https://github.com/adafruit/Adafruit_CircuitPython_Bundle/ will add automatic generation of these images so that they can also be embedded in the Learn website.

### CODE REVIEW

File: create_requirement_images.py [hunk 1]
```diff
@@ -40,20 +42,26 @@
...
-right_triangle = Image.open("img/right_triangle.png")
-down_triangle = Image.open("img/down_triangle.png")
 
-folder_icon = Image.open("img/folder.png")
-folder_hidden_icon = Image.open("img/folder_hidden.png")
-file_icon = Image.open("img/file.png")
-file_hidden_icon = Image.open("img/file_hidden.png")
-file_empty_icon = Image.open("img/file_empty.png")
-file_empty_hidden_icon = Image.open("img/file_empty_hidden.png")
+def asset_path(x):
```
Comment (FoamyGuy) [inline create_requirement_images.py]: Very minor, but would be good if we can use `asset_name` or `filename` or something more descriptive than `x` for the parameter here.

**Review by FoamyGuy — COMMENTED** [review]
I think I'm not understanding how the new mode is meant to operate. I tried running a command like this: 
```
 python create_requirement_images.py bundle adafruit-circuitpython-bundle-6.x-mpy-20210527/examples/turtle_swirl.py
```
But I end up with a different output image:

adafruit-circuitpython-bundle-6 x-mpy-20210527_turtle_swirl py

It looks like it grabbed all of the other directories inside of the examples folder in the bundle. Perhaps it's meant to be run on examples directories inside of individual repos rather than the full bundle itself?

**Comment by FoamyGuy** [comment]
After a bit more testing I do think it seems like it is meant to run on individual library repo I ran it like this:
```
python create_requirement_images.py bundle ../Adafruit_CircuitPython_Bitmap_Font/examples/bitmap_font_label_forkawesome.py
```
and I did get a requirement image that looks correct to me:
_Adafruit_CircuitPython_Bitmap_Font_bitmap_font_label_forkawesome py

Renaming the `x` parameter was the only other thing I found. I don't think that is necessarily a deal breaker either, If we want to merge without changing that, I am not fully opposed.

**Comment by jepler** [comment]
It's intended to run from the bundle source tree (e.g., Adafruit_CircuitPython_Bundle), like so:
```
find libraries -pattern "libraries/*/examples/*.py" \
    -exec python3 CircuitPython_Library_Screenshot_Maker/create_requirement_images_bundle.py {} +
```
so that it is invoked (more than once, if necessary) with each example listed on the commandline including its path.  So, more similar to how you invoked it in the second case.

If there are unrelated subdirectories present, those are included in the listing, whether or not they contain files that would actually be gathered up.  This problem exists both for Learn and for Bundle images, but may matter more with Bundle images.  For instance, the image for adafruit_io_simpletest shows the folders `adafruit_io_http` and `adafruit_io_mqtt` even though they are unnecessary:
adafruitio_adafruit_io_simpletest py

**Review by FoamyGuy — APPROVED** [review]
Neat! My "TerminalFu" is not quite to the level that I would have figured out how to use `find` to automate the whole list like that.

I can work this weekend on trying to filter out folders that are not used for the specific example being built. 

My first thought is to check if the code contains the folder name anywhere and not show it if not, this would not be guaranteed to be 100% error proof but should filter out things like the directories in the io library examples I think.

**Pull request merged**